fireatlas/FireMain.py

```diff
@@ -570,7 +570,12 @@ def Fire_Forward(tst: TimeStep, ted: TimeStep, restart=False, region=None, read_
         read_preprocessed(t, region=region, location=read_location)
         for t in list_of_ts
     ]
-    allpixels = pd.concat([df for df in list_of_allpixels if not df.empty])
+    non_empty_dfs = [df for df in list_of_allpixels if not df.empty]
+    if len(list_of_allpixels) > 0 and len(non_empty_dfs) == 0:
+        logger.warning(f"There are no new pixels for {ted}")
+        allpixels = list_of_allpixels[0]
```

Comment:
yeah, I see how hard this edge case is to reason about via the two paths of `t_saved -> list_of_ts` vs full `list_of_ts` 😏 

So this line `allpixels = list_of_allpixels[0]` assumes we are in the `t_saved -> list_of_ts` flow then b/c otherwise we'd at least have some non-empty df(s)? Is that right? 

And it just returns an empty df so that later on in the `t_saved` flow we can just `allpixels = pd.concat([allpixels_saved, allpixels])` concat it with the saved version and move on.

If I have that correct then looks fine. I should definitely write tests around this stuff

Reply:
We discussed out of band, but you can technically get to this spot without a `t_saved`. It's always valid to have no pixels in a preprocess_region_t dataframe (although sometimes it is suspicious #48)

It returns an empty df with the right structure which will either get concatted with the saved allpixels (in the case of `t_saved`) _or_ get fed into Fire_Forward and result in no fires (this is more likely for a small region).